src/vmware/azext_vmware/custom.py

```diff
@@ -177,13 +170,19 @@ def hcxenterprisesite_delete(cmd, client: AVSClient, resource_group_name, privat
 
 
 def datastore_create(cmd, client: AVSClient, resource_group_name, private_cloud, cluster, name, nfs_provider_ip=None, nfs_file_path=None, endpoints=[], lun_name=None):
-    from azext_vmware.vendored_sdks.avs_client.models import Datastore, NetAppVolume, DiskPoolVolume
-    datastore = Datastore()
-    if nfs_provider_ip is not None or nfs_file_path is not None:
-        datastore.net_app_volume = NetAppVolume(nfs_provider_ip=nfs_provider_ip, nfs_file_path=nfs_file_path)
-    if len(endpoints) > 0 or lun_name is not None:
-        datastore.disk_pool_volume = DiskPoolVolume(endpoints=endpoints, lun_name=lun_name)
-    return client.datastores.begin_create(resource_group_name=resource_group_name, private_cloud_name=private_cloud, cluster_name=cluster, datastore_name=name, datastore=datastore)
+    print('Please use "az vmware datastore netapp-volume create" or "az vmware datastore disk-pool-volume create" instead.')

```

Comment:
Are you sure we need to delete this logic now? This will lead to breaking change

In fact, our common deprecation process is as follows:
Maybe we can consider keeping the logic of creating datastore, and let users continue to use the command `az vmware datastore create` for a while. When the user executes this command, it will prompt the users this command will replaced by `az vmware datastore netapp-volume create` and `az vmware datastore disk-pool-volume create` in the future, but this command still works~ 
After users see the hint, then most people gradually use the new commands. At this time, we launch the breaking change, which will have less impact on users.

Do you think this process is suitable for this PR?

Reply:
I am sure that we need to delete this logic now.  There were breaking changes in the API from preview to stable. This command with have breaking changes. The "privateClouds/clusters/datastores" resource type is still behind a feature flag with a limited audience. cc @Raaghavi

Reply:
@ctaggart Does this mean that the users will not have the opportunity to use the preview API using this version of vmware extension?

Reply:
Correct. This pull request updates the CLI to stable AVS API version 2021-06-01.

Reply:
Got it. In that case, I would agree with @ctaggart. We do not have any customers using this API yet & we are behind a feature flag. We also plan on deprecating the preview API soon. So the breaking change introduced in this PR will have very limited impact.

Reply:
OK, got it~